apps/google-docs/src/components/ContentTypePickerModal.tsx

```diff
@@ -0,0 +1,161 @@
+  onClose: () => void;
+  onSelect: (contentTypes: SelectedContentType[]) => void;
+}
+
+export const ContentTypePickerModal = ({
+  sdk,
+  isOpen,
+  onClose,
+  onSelect,
+}: ContentTypePickerModalProps) => {
+  const [contentTypes, setContentTypes] = useState<ContentTypeProps[]>([]);
+  const [selectedContentTypes, setSelectedContentTypes] = useState<SelectedContentType[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    // Fetch content types when component mounts
+    const fetchContentTypes = async () => {
+      try {
+        setIsLoading(true);
+        const space = await sdk.cma.space.get({});
+        const environment = await sdk.cma.environment.get({ spaceId: space.sys.id });
+        const contentTypesResponse = await sdk.cma.contentType.getMany({
+          spaceId: space.sys.id,
+          environmentId: environment.sys.id,
+        });
+        setContentTypes(contentTypesResponse.items || []);
+      } catch (error) {
+        console.error('Failed to fetch content types:', error);
+        sdk.notifier.error('Failed to load content types');
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchContentTypes();
+  }, [sdk]);
+
+  useEffect(() => {
+    // Reset selection when modal opens
+    if (isOpen) {
+      setSelectedContentTypes([]);
+    }
+  }, [isOpen]);
+
+  const handleAddContentType = (contentTypeId: string) => {
+    if (!contentTypeId) return;
+
+    const contentType = contentTypes.find((ct) => ct.sys.id === contentTypeId);
+    if (contentType && !selectedContentTypes.some((ct) => ct.id === contentTypeId)) {
+      setSelectedContentTypes([
+        ...selectedContentTypes,
+        { id: contentType.sys.id, name: contentType.name },
+      ]);
+    }
+  };
+
+  const handleRemoveContentType = (contentTypeId: string) => {
+    setSelectedContentTypes(selectedContentTypes.filter((ct) => ct.id !== contentTypeId));
+  };
+
+  const handleContinue = () => {
+    if (selectedContentTypes.length === 0) {
+      sdk.notifier.error('Please select at least one content type');
+      return;
+    }
+
+    onSelect(selectedContentTypes);
+  };
+
+  const availableContentTypes = contentTypes.filter(
+    (ct) => !selectedContentTypes.some((selected) => selected.id === ct.sys.id)
+  );
+
+  return (
+    <Modal title="Select content type(s)" isShown={isOpen} onClose={onClose} size="medium">
+      {() => (
+        <>
+          <Modal.Header title="Select content type(s)" onClose={onClose} />
+          <Modal.Content>
+            <Paragraph marginBottom="spacingM">
+              Select the content type(s) you would like to use with this sync.
+            </Paragraph>
+            <FormControl isRequired>
+              <FormControl.Label>
+                Content type{' '}
+                {selectedContentTypes.length > 0 && (
+                  <span style={{ fontWeight: 'normal', color: '#536171' }}>(required)</span>
+                )}
```

Comment:
We could use F36 here to stick with our color tokens `<Text fontColor="red500">`